EXTENSION_POLICY.md

```diff
@@ -0,0 +1,57 @@
+# Envoy Extension Policy
+
+## Quality requirements
+
+All extensions contained in the main Envoy repository will be held to the same quality bar as the
+core Envoy code. This includes coding style, code reviews, test coverage, etc. In the future we
+may consider creating a sandbox repository for extensions that are not compiled/tested by default
+and held to a lower quality standard, but that is out of scope currently.
+
+## Adding new extensions
+
+The following procedure will be used when proposing new extensions for inclusion in the repository:
+  1. A GitHub issue should be opened describing the proposed extension as with any major feature
+  proposal.
+  1. All extensions must be sponsored by an existing maintainer. Sponsorship means that the
+  maintainer will shepherd the extension through design/code reviews. Maintainers can self-sponsor
+  extensions if they are going to write them, shepherd them, and maintain them.
+
+  Sponsorship serves two purposes.
+    - It ensures that the extension will ultimately meet the Envoy quality bar.
```

Comment:
add new line before this so it parsed by GitHub correctly. (same for other places below) https://github.com/envoyproxy/envoy/blob/dd05064b15aa3b7305e82b7954f142ce7ce705df/EXTENSION_POLICY.md